system-tests/e2e-testsupport/src/test/java/com/hotels/styx/servers/MockOriginServerTest.java

```diff
@@ -0,0 +1,144 @@
+        server.stop();
+        HttpsURLConnection.setDefaultHostnameVerifier(oldHostNameVerifier);
+    }
+
+    @Test
+    public void configuresEndpoints() throws Exception {
+        int adminPort = 43080;
+        int serverPort = 43081;
+
+        server = MockOriginServer.create("", "", adminPort, new HttpConnectorConfig(serverPort))
+                .start()
+                .stub(WireMock.get(urlMatching("/.*")), aResponse()
+                        .withStatus(200)
+                        .withHeader("a", "b")
+                        .withBody("Hello, World!"));
+
+        HttpResponse response = send(client,
+                get(format("http://localhost:%d/mock", serverPort))
+                        .header("X-Forwarded-Proto", "http")
+                        .build())
+                .responseBuilder()
+                .build();
+
+        assertThat(response.status(), is(OK));
+        assertThat(response.header("a"), is(Optional.of("b")));
+
+        server.verify(getRequestedFor(urlEqualTo("/mock"))
+                .withHeader("X-Forwarded-Proto", valueMatchingStrategy("http")));
+    }
+
+    @Test
+    public void configuresTlsEndpoints() throws Exception {
+        int adminPort = 43082;
+        int serverPort = 43083;
+
+        server = MockOriginServer.create("", "", adminPort,
+                new HttpsConnectorConfig.Builder()
+                        .port(serverPort)
+                        .build())
+                .start()
+                .stub(WireMock.get(urlMatching("/.*")), aResponse()
+                        .withStatus(200)
+                        .withHeader("a", "b")
+                        .withBody("Hello, World!"));
+
+        HttpResponse response = send(client,
+                get(format("https://localhost:%d/mock", serverPort))
+                        .header("X-Forwarded-Proto", "http")
+                        .build())
+                .responseBuilder()
+                .build();
+
+        assertThat(response.status(), is(OK));
+        assertThat(response.header("a"), is(Optional.of("b")));
+
+        server.verify(getRequestedFor(urlEqualTo("/mock"))
+                .withHeader("X-Forwarded-Proto", valueMatchingStrategy("http")));
+    }
+
+
+    private HttpResponse.DecodedResponse send(HttpClient client, HttpRequest request) {
+        return client.sendRequest(request)
+                .flatMap(response -> response.decode(bytebuf -> bytebuf.toString(UTF_8), 256 * 1024))
+                .toBlocking()
+                .first();
+    }
+
+    private ValueMatchingStrategy valueMatchingStrategy(String matches) {
+        ValueMatchingStrategy strategy = new ValueMatchingStrategy();
+        strategy.setMatches(matches);
+        return strategy;
+    }
+
+    private HostnameVerifier disableHostNameVerification() {
+        HostnameVerifier old = HttpsURLConnection.getDefaultHostnameVerifier();
+        HttpsURLConnection.setDefaultHostnameVerifier(new HostnameVerifier() {
```

Comment:
Please statically import `setDefaultHostnameVerifier`.
Also, the `HostnameVerifier` instantiated here can be a lambda.